Fix pot deposit handling

m-roberts opened the pull request:
I got [this](https://community.monzo.com/t/adding-money-to-pots-returns-an-error-that-the-value-must-be-more-than-0/108829/11) issue when trying to make a deposit into a pot.

I found that it was due to "[the fact the data was in the query string and not the body](https://community.monzo.com/t/adding-money-to-pots-returns-an-error-that-the-value-must-be-more-than-0/108829/12)".

This is likely an issue across the library, but the change offered in this PR is what I needed for my purposes.

m-roberts committed:
Fix pot deposit handling

m-roberts commented on src/pymonzo/resources.py:
```diff
@@ -43,7 +44,7 @@ def _get_response(
             MonzoAPIError: When Monzo API returned an error.
         """
-        response = getattr(self.client.session, method)(endpoint, params=params)
+        response = getattr(self.client.session, method)(endpoint, params=params, data=data)
```
**suggestion:** 
```suggestion
        kwargs = {"params": params}
        if method in ["post", "put", "delete"]:
            kwargs["data"] = data

        response = getattr(self.client.session, method)(endpoint, **kwargs)
```

m-roberts committed:
Update src/pymonzo/resources.py

m-roberts committed:
Update resources.py

pawelad commented:
Thanks for highlighting this, I'm honestly not sure how I haven't noticed it before. I'll check a couple of things, merge this and try to release a new version soon-ish.

pawelad committed:
DELETE does not allow form data, but PATCH does

pawelad committed:
Update `_get_response` docstring

pawelad committed:
Use 'form data' instead of 'query params' for relevant Monzo API endpoints

I got the list by looking through `--form` occurrences in https://docs.monzo.com/

pawelad committed:
Add CHANGELOG entry

pawelad committed:
Update tests
